src/editor/object_option.hpp

```diff
@@ -443,6 +443,21 @@ class ButtonOption : public ObjectOption
   ButtonOption& operator=(const ButtonOption&) = delete;
 };
 
+class StringArrayOption : public ObjectOption
+{
+public:
+  StringArrayOption(const std::string& text, const std::string& key, std::vector<std::string>* items);
+
```

Comment:
Is there a reason to pass a pointer there?

The problem with passing a pointer is that the caller will bear the responsibility of managing its lifetime, which prevents simple calls like `ctor({ "foo", "bar" }, ...)` or similar. Also, is it expected to pass a null pointer there?

I believe it would be preferable to use `const&` and to copy the vector into the member variable, if possible.